packages/cli/src/agent.ts

```diff
@@ -0,0 +1,487 @@
+const agentPushCommand = new MedplumCommand('push');
+const agentReloadConfigCommand = new MedplumCommand('reload-config');
+const agentUpgradeCommand = new MedplumCommand('upgrade');
+
+export const agent = new MedplumCommand('agent');
+addSubcommand(agent, agentStatusCommand);
+addSubcommand(agent, agentPingCommand);
+addSubcommand(agent, agentPushCommand);
+addSubcommand(agent, agentReloadConfigCommand);
+addSubcommand(agent, agentUpgradeCommand);
+
+agentStatusCommand
+  .description('Get the status of a specified agent')
+  .argument('[agentIds...]', 'The ID(s) of the agent(s) to get the status of')
+  .option(
+    '--criteria <criteria>',
+    'An optional FHIR search criteria to resolve the agent to get the status of. Mutually exclusive with [agentIds...] arg'
+  )
+  .addOption(
+    new Option('--output <format>', 'An optional output format, defaults to table')
+      .choices(['table', 'json'])
+      .default('table')
+  )
+  .action(async (agentIds, options) => {
+    await callAgentBulkOperation({
+      operation: '$bulk-status',
+      agentIds,
+      options,
+      parseSuccessfulResponse: (response: AgentBulkOpResponse<Parameters>) => {
+        const statusEntry = parseParameterValues(response.result, {
+          required: ['status', 'version'],
+          optional: ['lastUpdated'],
+        });
+
+        return {
+          id: response.agent.id as string,
+          name: response.agent.name,
+          enabledStatus: response.agent.status,
+          version: statusEntry.version,
+          connectionStatus: statusEntry.status,
+          statusLastUpdated: statusEntry.lastUpdated ?? 'N/A',
+        } satisfies StatusRow;
+      },
+    });
+  });
+
+agentPingCommand
+  .description('Ping a host from a specified agent')
+  .argument('<ipOrDomain>', 'The IPv4 address or domain name to ping')
+  .argument(
+    '[agentId]',
+    'Conditionally optional ID of the agent to ping from. Mutually exclusive with --criteria <criteria> option'
+  )
+  .option('--count <count>', 'An optional amount of pings to issue before returning results', '1')
+  .option(
+    '--criteria <criteria>',
+    'An optional FHIR search criteria to resolve the agent to ping from. Mutually exclusive with [agentId] arg'
+  )
+  .action(async (ipOrDomain, agentId, options) => {
+    const medplum = await createMedplumClient(options);
+    const agentRef = await resolveAgentReference(medplum, agentId, options);
+
+    const count = Number.parseInt(options.count, 10);
+    if (Number.isNaN(count)) {
+      throw new Error('--count <count> must be an integer if specified');
+    }
+
+    let pingResult: string;
+    try {
+      pingResult = (await medplum.pushToAgent(agentRef, ipOrDomain, `PING ${count}`, ContentType.PING, true, {
+        maxRetries: 0,
+      })) as string;
+    } catch (err) {
+      throw new Error('Unexpected response from agent while pinging', { cause: err });
+    }
+
+    console.info(pingResult);
```

Comment:
Suggest tucking statements inside `try` for cohesion:
```suggestion
    try {
      const pingResult = (await medplum.pushToAgent(agentRef, ipOrDomain, `PING ${count}`, ContentType.PING, true, {
        maxRetries: 0,
      })) as string;
      console.info(pingResult);
    } catch (err) {
      throw new Error('Unexpected response from agent while pinging', { cause: err });
    }
```